btor/src/boolector_term.cpp

```diff
@@ -267,9 +265,27 @@ Sort BoolectorTerm::get_sort() const
   return sort;
 }
 
+bool BoolectorTerm::is_symbol() const
+{
+  // functions, parameters, and symbolic constants are all symbols
+  auto bkind = bn->kind;
+  return !negated
+         && ((bkind == BTOR_VAR_NODE) || (bkind == BTOR_UF_NODE)
+             || (bkind == BTOR_PARAM_NODE));
+}
+
+bool BoolectorTerm::is_param() const
+{
+  return !negated && (bn->kind == BTOR_PARAM_NODE);
+}
+
 bool BoolectorTerm::is_symbolic_const() const
 {
-  return is_sym;
+  bool is_fun = boolector_is_fun(btor, node);
+  auto bkind = bn->kind;
+  bool is_sym =
+      !negated && ((bkind == BTOR_VAR_NODE) || (bkind == BTOR_UF_NODE));
+  return is_sym && !is_fun;
```

Comment:
maybe `return !is_fun && !is_param && is_symbol();`

Reply:
Made it simpler in e3e74f36278132d722bf3991c6515fa38de16c1b.